kyo-core/shared/src/main/scala/kyo/channels.scala

```diff
@@ -118,15 +118,17 @@ object Channels:
                                     None
                                 case r: Some[Seq[T]] =>
                                     def dropTakes(): Unit < IOs =
-                                        takes.poll() match
-                                            case null => ()
-                                            case p =>
-                                                p.interrupt.map(_ => dropTakes())
+                                        Loops.foreach {
+                                            takes.poll() match
+                                                case null => Loops.doneUnit
+                                                case p    => p.interrupt.map(_ => Loops.continueUnit)
+                                        }
                                     def dropPuts(): Unit < IOs =
-                                        puts.poll() match
-                                            case null => ()
-                                            case (_, p) =>
-                                                p.interrupt.map(_ => dropPuts())
+                                        Loops.foreach {
+                                            puts.poll() match
+                                                case null   => Loops.doneUnit
+                                                case (_, p) => p.interrupt.map(_ => Loops.continueUnit)
+                                        }
```

Comment:
These cases were already stack safe since `interrupt` always suspends with `IOs` but it's best to be on the safe side since the code can change over time and `Loops` introduces little overhead.